streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableOuterJoin.java

```diff
@@ -68,14 +60,20 @@ public void init(ProcessorContext context) {
         }
 
         @Override
-        public void process(K key, V1 value1) {
+        public void process(K key, Change<V1> change) {
             V newValue = null;
+            V oldValue = null;
             V2 value2 = valueGetter.get(key);
 
-            if (value1 != null || value2 != null)
-                newValue = joiner.apply(value1, value2);
+            if (change.newValue != null || value2 != null)
+                newValue = joiner.apply(change.newValue, value2);
+
+            if (sendOldValues) {
```

Comment:
You mean `if (sendOldValues && (change.oldValue != null || value2 != null))`?
I was just making it clear and easy to read that this branch is for the case sending old values rather than having a complicated if-condition.